src/lightning_classes/lightning_image_classification.py

```diff
@@ -13,22 +13,14 @@ def __init__(self, cfg: DictConfig):
         self.cfg = cfg
         self.model = load_obj(cfg.model.class_name)(cfg=cfg)
         self.loss = load_obj(cfg.loss.class_name)()
-        self.metrics = [
-            {
-                'metric': load_obj(self.cfg.metric.metric.class_name)(**cfg.metric.metric.params).to(
-                    self.cfg.general.device
-                ),
-                'metric_name': self.cfg.metric.metric.metric_name,
-            }
-        ]
+        self.metrics = torch.nn.ModuleDict({
+            self.cfg.metric.metric.metric_name: load_obj(self.cfg.metric.metric.class_name)(**cfg.metric.metric.params)
+        })
         if 'other_metrics' in self.cfg.metric.keys():
             for metric in self.cfg.metric.other_metrics:
-                self.metrics.append(
-                    {
-                        'metric': load_obj(metric.class_name)(**metric.params).to(self.cfg.general.device),
-                        'metric_name': metric.metric_name,
-                    }
-                )
+                self.metrics.update({
```

Comment:
If `self.metrics` is changed, than it is necessary to update the code for logging metrics in train and validation steps.